modules/images/webp-uploads/load.php

```diff
@@ -160,33 +271,22 @@ function webp_uploads_generate_image_size( $attachment_id, $size, $mime ) {
 		return new WP_Error( 'image_mime_type_not_supported', __( 'The provided mime type is not supported.', 'performance-lab' ) );
 	}
 
-	$extension = explode( '|', $allowed_mimes[ $mime ] );
-	$extension = reset( $extension );
+	$height = isset( $size_data['height'] ) ? (int) $size_data['height'] : 0;
+	$width  = isset( $size_data['width'] ) ? (int) $size_data['width'] : 0;
+	$crop   = isset( $size_data['crop'] ) && $size_data['crop'];
 
-	$width  = null;
-	$height = null;
-	$crop   = false;
-
-	if ( isset( $metadata['sizes'][ $size ]['width'] ) ) {
-		$width = $metadata['sizes'][ $size ]['width'];
-	} elseif ( isset( $sizes[ $size ]['width'] ) ) {
-		$width = $sizes[ $size ]['width'];
+	if ( $width <= 0 && $height <= 0 ) {
+		return new WP_Error( 'image_wrong_dimensions', __( 'At least one of the dimensions must be a positive number.', 'performance-lab' ) );
 	}
```

Comment:
I think this is not fully correct, both dimensions must be positive numbers, not just one. Something like `300x0` or `0x300` wouldn't make sense either.
```suggestion
	if ( $width <= 0 || $height <= 0 ) {
		return new WP_Error( 'image_wrong_dimensions', __( 'Both image dimensions must be positive numbers.', 'performance-lab' ) );
	}
```

Reply:
Actually having an image with only one dimension is a valid image size like: 

-`300x0`

Specifically, this is used in cases when you want to preserve the aspect ratio of an image by just setting the width in most cases so the height would be 0 or can be specified as 0.

https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/media.php#L535

Reply:
Ahh I wasn't aware that works, I thought it would try to make it actually 300x0 then. In that case, all good!